src/traits.rs

```diff
@@ -13,17 +14,23 @@ use typenum::Unsigned;
 /// # Safety
 ///
 /// `ArrayType` MUST be an array with a number of elements exactly equal to
-/// [`Unsigned::USIZE`]. Breaking this requirement will cause undefined behavior.
-///
-/// NOTE: This trait is effectively sealed and can not be implemented by third-party crates.
-/// It is implemented only for a number of types defined in [`typenum::consts`].
-pub unsafe trait ArraySize: Unsigned {
```

Comment:
We use `Unsigned::USIZE` quite a bit, FWIW. I guess all of these usages would hypothetically have to change to `ArraySize::Size::USIZE`

Reply:
Yeah, that goes for my follow up suggestion of adding the convenience const. Although you do run into ambiguities then